scheduler/ecs/ecs.go

```diff
@@ -288,6 +278,41 @@ func (m *Scheduler) Instances(ctx context.Context, appID string) ([]*scheduler.I
 	return instances, nil
 }
 
+func (m *Scheduler) describeTaskDefinitions(ctx context.Context, tasks []*ecs.Task) (map[string]*ecs.TaskDefinition, error) {
+	taskDefinitions := make(map[string]*ecs.TaskDefinition)
+
+	for _, t := range tasks {
+		k := *t.TaskDefinitionArn
+		taskDefinitions[k] = nil
+	}
+
+	ch := make(chan struct {
+		resp *ecs.DescribeTaskDefinitionOutput
+		err  error
+	}, len(taskDefinitions))
+	for arn := range taskDefinitions {
+		go func(arn string) {
+			resp, err := m.ecs.DescribeTaskDefinition(ctx, &ecs.DescribeTaskDefinitionInput{
+				TaskDefinition: aws.String(arn),
+			})
+			ch <- struct {
+				resp *ecs.DescribeTaskDefinitionOutput
+				err  error
+			}{resp, err}
+		}(arn)
+	}
+
+	for arn := range taskDefinitions {
+		result := <-ch
+		if err := result.err; err != nil {
```

Comment:
I think with https://github.com/remind101/empire/pull/781, we're less likely to hit throttling even if we parallelize like this. With my test program, I had to do about 150 DescribeTaskDefinitions in parallel until I got throttled.

It would definitely still be possible if you did something like `while true; do emp ps; done`, but in that case all we can really do us increase the number of retries in the AWS client.